lm_api_specs.md

```diff
@@ -1,35 +1,47 @@
-tests.  On the other hand, Registries can *act on their users' behalf*, and thus
-can manage workflows and workflow tests that belong to their users. For
+1. **Users**
+2. **Registries**
+
+Both can manage their own workflows and associated workflow tests through the LM API. In addition, Registries can _act on their users' behalf_, and thus, they can manage workflows and workflow tests that belong to their users. For
 instance, you may register a workflow on the [Workflow Hub
 registry](https://workflowhub.eu/) and it may in turn register on your behalf
 the workflow's tests with the Life Monitor.
 
-The type of client has implications on the authentication method to be used and
-on the parts of the API that are relevant.
+### Clients
+
+Thanks to the RESTful implementation and the adoption of OAuth 2.0 as authorisation protocol, the LM API provides support for a variety of different clients (e.g., scripting tools, web apps, etc.).
+
+Clients allow users and registries to interact with the LifeMonitor API.
+With respect to the actors they support, we can distinguish between:
+
+- **Generic Clients**: can act on behalf of a user to register user's workflows and associated workflow tests;
+
+- **Registry Clients**: allow a registry to submit its own workflows and associated workflow tests. In addition, registries can act on their users’ behalf and submit their users' workflows and associated workflow tests.
+
+<img alt="Life Monitor client types" src="images/lm_clients.png" width="900" />
+
+The authorisation mechanisms available for clients are:
 
-* **User clients**:
-  * must authenticate with API Key or OAuth AuthorizationCodeFlow;
-  * can't use Registry Client API resources.
-* **Registry clients**:
-  * must authenticate with OAuth RegistryClientCredentials or OAuth RegistryCodeFlow;
-  * can't use User Client API resources.
+* **API key**, can be used to implement generic clients;
+* **OAuth2 Client Credentials**, only available for trusted workflow registries (see *RegistryClientCredentials*);
+* **OAuth2 Authorization Code**, available for both generic clients (see *AuthorizationCodeFlow*) and registry clients (see *RegistryCodeFlow*).
 
+Clients can query API endpoints accordingly to the authorisation mechanism they adopt.
+There are in fact resources which are _"contextual"_ to the actor which the client is acting on behalf of and the type of "impersonated" actor is determined by the authorisation grant in use. Examples of contextual resources are:
 
-Follow [this link to see the API documentation and examples](https://api-lifemonitor-dev.rahtiapp.fi/static/apidocs.html).  You can find the raw YAML API specifications in [the source code repository](https://github.com/crs4/life_monitor/tree/master/specs).  
+- `/users/current/*`, which assume an authenticated _user_ and thus can be queried only by clients authorised through an _API key_ or an _OAuth2 Authorization Code flow_;
```

Comment:
Looking at the specs, I think that currently `/users/current` and `/users/current/workflows` are also accessible via RegistryCodeFlow